File: arcade/intro/python/almostIncreasingSequence.py
def almostIncreasingSequence(sequence):
    inc = False
    nb = 0
    while not inc:
        if nb > 1:
            return False
        inc = True
        i = 0
        while i < len(sequence)-1:
            if sequence[i] > sequence[i+1]:
                inc = False
                try:    
                    if sequence[i] < sequence[i+2]:
                        del sequence[i+1]
                        nb += 1
                        break
                except:
                    pass
                try:    
                    if (i == len(sequence)-2) and sequence[i] > sequence[i+1]:
                        del sequence[i+1]
                        nb += 1
                        break
                except:
                    pass
                del sequence[i]
                nb += 1
                break
            elif sequence[i] == sequence[i+1]:
                inc = False
                del sequence[i]
                nb +=1
                break
            else:
                inc = True
            i += 1
    if nb > 1:
        return False
    return True

File: arcade/intro/C/almostIncreasingSequence.c
bool almostIncreasingSequence(arr_integer sequence) {
	int arrayLength=sequence.size;
	int i = 0;
	int count = 0;
	int flag = false;
	int max = -100000;
	int secondMax = -100000;

	for (i = 0; i < arrayLength; i++) {
		if (sequence.arr[i] > max) {
			secondMax = max;
			max = sequence.arr[i];
		} else if (sequence.arr[i] > secondMax) {
			if (flag) {
				max = sequence.arr[i];
				flag = false;
			} else {
				secondMax = sequence.arr[i];
				flag = true;
				count++;
			}
		} else {
			count++;
		}
	}


	if (count == 1) {
		return true;
	} else {
		return false;
	}

}
